src/components/GameView/ChooseTwoDialog.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <base-dialog v-if="oneOff" v-model="show" id="choose-two-dialog" title="Choose Two">
+    <template #body>
+      <p class="mb-4">Which Two would you like to counter with? (Click the card)</p>
+      <div id="twos-in-hand" class="d-flex justify-center mb-4">
+        <game-card
+          v-for="two in twosInHand"
+          :key="two.id"
+          :suit="two.suit"
+          :rank="two.rank"
+          :data-counter-dialog-card="`${two.rank}-${two.suit}`"
+          @click="counter(two)"
+        />
+      </div>
+    </template>
+    <template #actions>
+      <v-btn variant="text" color="surface-1" data-cy="cancel-counter" @click="resolve">
+        Cancel
+      </v-btn>
+    </template>
+  </base-dialog>
+</template>
+
+<script>
+import GameCard from '@/components/GameView/GameCard.vue';
+import BaseDialog from '@/components/Global/BaseDialog.vue';
+
+export default {
+  name: 'ChooseTwoDialog',
+  components: {
+    BaseDialog,
+    GameCard,
+  },
+  emits: ['counter', 'resolve'],
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+    oneOff: {
+      type: Object,
+      requried: true,
+    },
+    // list of card objects for available twos
+    twosInHand: {
+      type: Array,
+      required: true,
+    },
+    twosPlayed: {
+      type: Array,
+      required: true,
+    },
+  },
+  computed: {
+    show: {
+      get() {
+        return this.modelValue;
+      },
+      set() {
+        // do nothing - parent controls whether dialog is open
+      },
+    },
+    opponentLastTwo() {
+      return this.twosPlayed && this.twosPlayed.length > 0
```

Comment:
`this.twosPlayed && this.twosPlayed.length` is being used a couple times, might want to make that a computed prop though the duplication is hardly a breaker.

Reply:
DRY'd this up by pulling that expression into a computed prop as you suggested